### PR TITLE
Add data source for WAF Groups

### PR DESCRIPTION
**Pull request by XaF**

### CODE REVIEW

**Comment by jacobbednarz** [comment]
Are you able to provide a description of what the use case is for this? I can't think of any and would like to make sure we have a use case before just adding these new datasources.

**Comment by XaF** [comment]
We want to be able to use the values returned by those data sources to use `for_each` on a resource element after and run bulk options for "most of" the groups/packages (for the other PR).

For instance, something like:

```terraform
resource "cloudflare_waf_group" "bulk" {
  for_each = {
    for _, group in data.cloudflare_waf_groups.selected_bulk.groups :
    group['id'] => group
    if group['id'] != xxxx // If I want to exclude this one here because I did a singular action on it
  }

  zone_id = data.cloudflare_waf_groups.selected_bulk.zone_id
  group_id = each.key
  package_id = each.value['package_id']
  mode = "off"
}
```

cc @sbfaulkner

File: cloudflare/data_source_waf_groups.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+)
+
+func dataSourceCloudflareWAFGroups() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareWAFGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
Comment (patryk) [inline cloudflare/data_source_waf_groups.go]: This field is required, yet not documented.

Comment (XaF) [reply cloudflare/data_source_waf_groups.go]: Fixed! As well as for the `package_id` field.

**Comment by XaF** [comment]
> Please also add new documentation page to TOC: https://github.com/terraform-providers/terraform-provider-cloudflare/blob/master/website/cloudflare.erb

Done!

**Comment by patryk** [comment]
Thanks!

**Comment by XaF** [comment]
Thank you! :)

**Pull request merged**